internal/graph/object_providers_test.go

```diff
@@ -119,10 +119,6 @@ func TestRecursiveObjectProvider(t *testing.T) {
 }
 
 func TestRecursiveTTUObjectProvider(t *testing.T) {
```

Comment:
why do we remove the leak check?  If it is because of the fan_in iterator, can we add them as [IgnoreTopFunction](https://pkg.go.dev/go.uber.org/goleak#IgnoreTopFunction)?